platform/osx/app/MapDocument.m

```diff
@@ -457,8 +488,8 @@ - (BOOL)validateMenuItem:(NSMenuItem *)menuItem {
     if (menuItem.action == @selector(dropManyPins:)) {
         return YES;
     }
-    if (menuItem.action == @selector(removeAllPins:)) {
-        return self.mapView.annotations.count;
+    if (menuItem.action == @selector(removeAllAnnotations:)) {
+        return self.mapView.annotations.count ? YES : NO;
```

Comment:
Switched to your suggestion in https://github.com/mapbox/mapbox-gl-native/commit/21ce6fbdf96e80272fa0893640f7da85497aa9aa#diff-1478e1e346e69c2437356a66353ddbf8R492.